src/header/common/content_type.rs

```diff
@@ -94,4 +99,20 @@ impl ContentType {
     }
 }
 
+#[cfg(feature = "serde-serialization")]
```

Comment:
Everything else looks great, but can we move this into the definition of the `header` macro, so that we automatically get serialization for most headers?